plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -972,7 +972,9 @@ private static ReaderPageSourceWithRowPositions createParquetPageSource(
 
             MessageType requestedSchema = new MessageType(fileSchema.getName(), parquetFields.stream().filter(Objects::nonNull).collect(toImmutableList()));
             Map<List<String>, ColumnDescriptor> descriptorsByPath = getDescriptors(fileSchema, requestedSchema);
-            TupleDomain<ColumnDescriptor> parquetTupleDomain = getParquetTupleDomain(descriptorsByPath, effectivePredicate);
+            // Parquet predicate pushdown is only supported for base columns (https://github.com/trinodb/trino/issues/9928)
+            TupleDomain<IcebergColumnHandle> baseColumnPredicate = effectivePredicate.filter((columnHandle, domain) -> columnHandle.isBaseColumn());
```

Comment:
Let's move this inside `getParquetTupleDomain` and make similar change to `ParquetPageSourceFactory#getParquetTupleDomain`. I think Hudi might also get this bug if it implements dereference pushdown.